src/structures/CommandInteractionOptionResolver.js

```diff
@@ -0,0 +1,169 @@
+     * @type {Client}
+     * @readonly
+     */
+    Object.defineProperty(this, 'client', { value: client });
+
+    /**
+     * The interaction options array.
+     * @type {CommandInteractionOption[]}
+     * @private
+     */
+    this._options = options ?? [];
+  }
+
+  /**
+   * Gets an option by its name.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?CommandInteractionOption} The option, if found.
+   */
+  get(name, required = false) {
+    const option = this._options.find(opt => opt.name === name);
+    if (option) {
+      return option;
+    } else if (required) {
+      throw new TypeError('COMMAND_INTERACTION_OPTION_NOT_FOUND', name);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Gets an option by name and property and checks its type.
+   * @param {string} name The name of the option.
+   * @param {ApplicationCommandOptionType[]} types The type of the option.
+   * @param {string[]} properties The property to check for for `required`.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?CommandInteractionOption} The option, if found.
+   * @private
+   */
+  _getTypedOption(name, types, properties, required) {
+    const option = this.get(name, required);
+    if (option) {
+      if (!types.includes(option.type)) {
+        throw new TypeError('COMMAND_INTERACTION_OPTION_TYPE', name, option.type, types);
+      } else if (required && properties.every(prop => option[prop] === null || typeof option[prop] === 'undefined')) {
+        throw new TypeError('COMMAND_INTERACTION_OPTION_EMPTY', name, option.type);
+      } else {
+        return option;
+      }
+    } else {
+      return option;
+    }
+  }
+
+  /**
+   * Gets a sub-command or sub-command group.
+   * @param {string} name The name of the sub-command or sub-command group.
+   * @returns {?CommandInteractionOptionResolver}
+   * A new resolver for the sub-command/group's options, or null if empty
+   */
+  getSubCommand(name) {
+    const option = this._getTypedOption(name, ['SUB_COMMAND', 'SUB_COMMAND_GROUP'], ['options'], false);
+    return option ? new CommandInteractionOptionResolver(this.client, option.options) : null;
+  }
+
+  /**
+   * Gets a boolean option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?boolean} The value of the option, or null if not set and not required.
+   */
+  getBoolean(name, required = false) {
+    const option = this._getTypedOption(name, ['BOOLEAN'], ['value'], required);
+    return option?.value ?? null;
+  }
+
+  /**
+   * Gets a channel option.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?GuildChannel|APIInteractionDataResolvedChannel}
```

Comment:
```suggestion
   * @returns {?(GuildChannel|APIInteractionDataResolvedChannel)}
```

Reply:
These types are the same

Reply:
I guess I should have explained why, the reasoning for this is explained here: #6021